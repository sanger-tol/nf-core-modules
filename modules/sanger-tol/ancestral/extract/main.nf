process ANCESTRAL_EXTRACT {
    tag "$meta.id"
    label 'process_low'

    // Docker image available at the project github repository
    container "ghcr.io/sanger-tol/busco_painter:1.0.1"

    input:
    tuple val(meta), path(fulltable)
    path(ancestraltable)

    output:
    tuple val(meta), path("*buscopainter_complete_location.tsv")  , emit: comp_location
    tuple val(meta), path("*buscopainter_duplicated_location.tsv"), emit: dup_location
    tuple val(meta), path("*summary.tsv")                         , emit: summary
    path "versions.yml"                                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "ANCESTRAL_EXTRACT module does not support Conda. Please use Docker / Singularity / Podman instead."
    }

    def args    = task.ext.args     ?: ''
    def prefix  = task.ext.prefix   ?: "${meta.id}"

    """
    buscopainter.py -r $ancestraltable -q $fulltable

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(echo \$(python3 --version 2>&1) | sed 's/^.*python //; s/Using.*\$//')
        buscopainter.py: \$(buscopainter.py -v)
    END_VERSIONS
    """

    stub:
    def prefix  = task.ext.prefix   ?: "${meta.id}"
    """
    touch ${prefix}_buscopainter_complete_location.tsv
    touch ${prefix}_buscopainter_duplicated_location.tsv
    touch ${prefix}_summary.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(echo \$(python --version 2>&1) | sed 's/^.*python //; s/Using.*\$//')
        buscopainter.py: \$(buscopainter.py -v)
    END_VERSIONS
    """
}
