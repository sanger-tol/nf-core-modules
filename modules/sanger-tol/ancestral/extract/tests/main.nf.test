
nextflow_process {

    name "Test Process ANCESTRAL_EXTRACT"
    script "../main.nf"
    process "ANCESTRAL_EXTRACT"

    tag "modules"
    tag "modules_sangertol"
    tag "ancestral"
    tag "ancestral/extract"

    test("GCA_963859965.1 - tsv") {

        when {

            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test-full-table' ], // meta map
                        file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/GCA_963859965.1_busco_full_table.tsv', checkIfExists: true)
                    ]
                )

                input[1] = Channel.of(
                    file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/Merian_elements_full_table.tsv', checkIfExists: true)
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("GCA_963859965.1 - tsv- stub") {

        options "-stub"

        when {

            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test-full-tabl' ], // meta map
                        file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/GCA_963859965.1_busco_full_table.tsv', checkIfExists: true)
                    ]
                )

                input[1] = Channel.of(
                    file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/Merian_elements_full_table.tsv', checkIfExists: true)
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
