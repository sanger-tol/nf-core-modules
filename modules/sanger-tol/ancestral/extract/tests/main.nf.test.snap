{
    "GCA_963859965.1 - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test-full-table"
                        },
                        "buscopainter_complete_location.tsv:md5,92a2d60e7a66edeb5070e3d982a53193"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test-full-table"
                        },
                        "buscopainter_duplicated_location.tsv:md5,23301496a69958b2b64d30df982ca2df"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test-full-table"
                        },
                        "buscopainter_summary.tsv:md5,16931cc78cc8506863b360e149d81111"
                    ]
                ],
                "3": [
                    "versions.yml:md5,92e1d26a59fe0815ee4e65a414349bf2"
                ],
                "comp_location": [
                    [
                        {
                            "id": "test-full-table"
                        },
                        "buscopainter_complete_location.tsv:md5,92a2d60e7a66edeb5070e3d982a53193"
                    ]
                ],
                "dup_location": [
                    [
                        {
                            "id": "test-full-table"
                        },
                        "buscopainter_duplicated_location.tsv:md5,23301496a69958b2b64d30df982ca2df"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test-full-table"
                        },
                        "buscopainter_summary.tsv:md5,16931cc78cc8506863b360e149d81111"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,92e1d26a59fe0815ee4e65a414349bf2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-06T11:47:09.348481052"
    },
    "GCA_963859965.1 - tsv- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test-full-tabl"
                        },
                        "test-full-tabl_[:]_buscopainter_complete_location.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test-full-tabl"
                        },
                        "test-full-tabl_[:]_buscopainter_duplicated_location.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test-full-tabl"
                        },
                        "test-full-tabl_[:]_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,92e1d26a59fe0815ee4e65a414349bf2"
                ],
                "comp_location": [
                    [
                        {
                            "id": "test-full-tabl"
                        },
                        "test-full-tabl_[:]_buscopainter_complete_location.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "dup_location": [
                    [
                        {
                            "id": "test-full-tabl"
                        },
                        "test-full-tabl_[:]_buscopainter_duplicated_location.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test-full-tabl"
                        },
                        "test-full-tabl_[:]_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,92e1d26a59fe0815ee4e65a414349bf2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-08T13:45:15.388341483"
    }
}